x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractorTests.java

```diff
@@ -442,6 +470,204 @@ public void testFieldsWithSingleValueArrayAsSubfield_TwoNestedLists2() {
         assertEquals(value, fe.extractFromSource(map));
     }
 
+    // "a": null
+    public void testMultiValueNull() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("a", null);
+        FieldHitExtractor fea = getArrayFieldHitExtractor("a", INTEGER);
+        assertEquals(singletonList(null), fea.extractFromSource(map));
+    }
+
+    // "a": [] / missing
+    public void testMultiValueEmpty() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("a", emptyList());
+        FieldHitExtractor fea = getArrayFieldHitExtractor("a", INTEGER);
+        assertEquals(emptyList(), fea.extractFromSource(map));
+
+        FieldHitExtractor feb = getArrayFieldHitExtractor("b", INTEGER);
+        assertEquals(emptyList(), feb.extractFromSource(map));
+    }
+
+    // "a": [i1, i2, ..]
+    public void testMultiValueImmediateFromMap() {
+        String fieldName = randomAlphaOfLength(5);
+        Map<String, Object> map = new HashMap<>();
+        List<Integer> list = randomList(2, 10, ESTestCase::randomInt);
+        map.put(fieldName, list);
+        FieldHitExtractor fe = getArrayFieldHitExtractor(fieldName, INTEGER);
+        assertEquals(list, fe.extractFromSource(map));
+    }
+
+    // "a": [i1, i2, ..]
```

Comment:
example doesn't show the difference from the prev test.

Reply:
Extended comment.